integration-cli/docker_cli_network_unix_test.go

```diff
@@ -1417,8 +1418,18 @@ func (s *DockerNetworkSuite) TestDockerNetworkInternalMode(c *check.C) {
 	c.Assert(waitRun("first"), check.IsNil)
 	dockerCmd(c, "run", "-d", "--net=internal", "--name=second", "busybox", "top")
 	c.Assert(waitRun("second"), check.IsNil)
-	_, _, err := dockerCmdWithError("exec", "first", "ping", "-c", "1", "www.google.com")
-	c.Assert(err, check.NotNil)
-	_, _, err = dockerCmdWithError("exec", "second", "ping", "-c", "1", "first")
+	errchan := make(chan error)
+	cmd := exec.Command(dockerBinary, "exec", "first", "ping", "-c", "1", "www.google.com")
+	go func() {
+		errchan <- cmd.Run()
+		close(errchan)
+	}()
+	select {
+	case <-time.After(time.Second):
+		cmd.Process.Kill()
+	case err := <-errchan:
+		c.Assert(err, check.NotNil)
+	}
```

Comment:
Don't we already have an utility function for that ? `DockerCmdWithTimeout` and/or `RunCommandWithOutputAndTimeout` (in `pkg/integration`).